src/common/io/io_windows.c

```diff
@@ -37,17 +37,26 @@ bool ffAppendFDBuffer(HANDLE handle, FFstrbuf* buffer)
     LARGE_INTEGER fileSize;
     if(!GetFileSizeEx(handle, &fileSize))
         fileSize.QuadPart = 0;
-
-    ffStrbufEnsureFree(buffer, fileSize.QuadPart > 0 ? (uint32_t)fileSize.QuadPart : 31);
+    
+    if (fileSize.QuadPart > 0)
+    {
+        // optimize for files has a fixed length,
+        // file can be very large, only keep necessary memory to save time and resources.
+        ffStrbufEnsureFixedLengthFree(buffer, (uint32_t)fileSize.QuadPart);
+    }
+    else
+        ffStrbufEnsureFree(buffer, 31);
     uint32_t free = ffStrbufGetFree(buffer);
+    int64_t remain = fileSize.QuadPart; // Windows does not have ssize_t, use int64_t to instead.
```

Comment:
> Windows does not have ssize_t, use int64_t to instead

Source?

Reply:
https://learn.microsoft.com/en-us/windows/win32/winprog/windows-data-types#ssize_t

Windows has SSIZE_T and the defination is actually difference from the *nix.

I'll try this: https://github.com/ycm-core/YouCompleteMe/issues/3949#issuecomment-944786867

Reply:
Did you search ssize_t in fastfetch code base?

Reply:
Oh, there is already a ssize_t in io_windows.c, i'm sorry :P